source/grid/grid_out.cc

```diff
@@ -3193,21 +3231,20 @@ namespace internal
           q_projector = new Quadrature<dim> (QProjector<dim>::project_to_all_faces(quadrature));
         }
 
+      std::vector<Point<spacedim>> point_cache;
       for (; cell!=endc; ++cell)
         {
           if (gnuplot_flags.write_cell_numbers)
             out << "# cell " << cell << '\n';
 
-          if (mapping==nullptr ||
-              (!cell->at_boundary() && !gnuplot_flags.curved_inner_cells))
+          if (mapping == nullptr ||
+              (dim == spacedim ? (!cell->at_boundary() && !gnuplot_flags.curved_inner_cells) :
+               // ignore checking for boundary or interior cells in the codim 1 case
+               false))
```

Comment:
this may not be the most obvious way to write this condition, but I guess that's ok :-)